### PR TITLE
Core: couple the overflow axes and fix mixed-overflow clipping

### PR DESCRIPTION
**Pull request by kane50613**
## Why

Two bugs in the same area, found while checking takumi's overflow handling against Blink.

**The axes were never coupled.** `resolve_overflows` read `overflow-x` and `overflow-y` straight off the computed style, so `overflow-x: hidden` next to `overflow-y: visible` stayed mixed. CSS Overflow 3 says a `visible` axis paired with one that is neither `visible` nor `clip` computes to a scrolling value instead, which is what `StyleAdjuster::AdjustOverflow` implements in Blink (`style_adjuster.cc:694-716`) and why Chrome clips both axes there.

This also meant the two backends disagreed. `takumi-svg` checks `clips_overflow()`, which is axis-agnostic, so it has always clipped both axes; only the raster path clipped one. The regenerated `.svg` for the affected fixture is unchanged, which is the evidence: SVG was already producing the corrected output.

**The mask fast path misread its sentinel.** `clip` next to `visible` is a legal pair and survives the coupling, so it still reaches the mask builder as a mixed state. `prepare_node_mask` marks an unclipped axis with `u32::MAX`, and `fill_rect_mask` narrowed it with `as i32`, which truncates to `-1` rather than saturating. The clip rectangle came out empty and the node rendered nothing at all. The rotated path never had this bug because `transformed_mask_point` compares in `u32`.

The existing fixture missed it because its box sits in a centering flex container, so the transform is a translation and the identity fast path never ran.

## What

- `resolve_overflows` applies the axis coupling. Every consumer already funnels through it: the taffy style conversion, the mask builder, the stacking context, and the scene walker. `clip` next to `visible` passes through untouched.
- `fill_rect_mask` clamps into `i32` range before narrowing.
- New fixture `style_overflow_clip_visible_identity_transform`: block root, padding, `clip`/`visible`. Without the mask fix it renders an empty border box with all the text gone.

## Fixture changes

`style_overflow_hidden_visible_text.webp` now clips vertically as well. That fixture is `Overflow::Hidden, Overflow::Visible`, exactly the pair the coupling rule targets. Its `.svg` is untouched. `style_overflow_clip_visible_image` is `Overflow::Clip, Overflow::Visible` and is untouched, confirming the rule only fires where it should.

takumi has no `auto`, so the coupling resolves to `hidden`. Without a scrolling box the two are visually identical, and `hidden` avoids reserving a scrollbar gutter in taffy.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved CSS overflow handling so paired horizontal/vertical values resolve consistently, avoiding “mixed” results when they should normalize.
  - Fixed rendering/clipping edge cases involving unbounded/sentinel values that could produce empty clip regions.
  - Added a helper to correctly classify `clip` vs `visible` overflow behavior.
- **Tests**
  - Added regression assertions for overflow resolution and mixed-axis rendering, including identity transform behavior.
- **Documentation**
  - Updated documentation with the overflow normalization and clipping fixes.

### CODE REVIEW

File: takumi-core/src/style/stylesheets_query.rs [hunk 1]
```diff
@@ -162,9 +162,19 @@ impl ComputedStyle {
...
+  /// <https://drafts.csswg.org/css-overflow-3/#overflow-properties>. Blink
+  /// resolves it to `auto`; without a scrolling box that is `hidden` here.
   pub fn resolve_overflows(&self) -> SpacePair<Overflow> {
-    SpacePair::from_pair(self.overflow_x, self.overflow_y)
+    let (x, y) = match (self.overflow_x, self.overflow_y) {
+      (Overflow::Visible, Overflow::Hidden) | (Overflow::Hidden, Overflow::Visible) => {
+        (Overflow::Hidden, Overflow::Hidden)
+      }
+      pair => pair,
+    };
```
[the remark this answers is not in the document]
Comment (kane50613) [reply takumi-core/src/style/stylesheets_query.rs:174]: Adopted the predicate form in 4d531726, as a named `Overflow::is_clip_or_visible` mirroring Blink's `IsOverflowClipOrVisible` rather than an inline `matches!` at both arms.

The `visible/auto` and `visible/scroll` coverage you asked for cannot be written: `Overflow` has exactly three variants (`Visible`, `Clip`, `Hidden`), so those pairs are unrepresentable and nothing currently falls through. The change is still worth making because the enum is `#[non_exhaustive]` and naming the value instead of the rule would silently skip the coupling if a variant is added later. Locked the classification with a unit test in `overflow.rs` instead.

Your structure also caught something I had wrong: Blink adjusts only the visible axis and leaves its companion alone, where I collapsed both to `hidden`. Those agree while `hidden` is the only value that reaches the rule, but the predicate form is correct in general. Full suite passes with no fixture movement, confirming the refactor is behaviour-identical today.
